### PR TITLE
Various compatibility updates

### PR DESCRIPTION
**Pull request by jamiesnape**
Relates https://github.com/RobotLocomotion/spartan/issues/266. PTAL @patmarion.

### CODE REVIEW

**Comment by patmarion** [comment]
Thanks @jamiesnape.  Looks like this PR failed the ubuntu14 tests.  While I would be happy to drop Ubuntu 14 at this point, I think it is easy to fix this failure by modifying `externals.cmake`.  See this line:

https://github.com/RobotLocomotion/director/blob/master/distro/superbuild/cmake/externals.cmake#L154

We need a similar line here:

https://github.com/RobotLocomotion/director/blob/master/distro/superbuild/cmake/externals.cmake#L583

Basically what is happening, is for systems without cmake < v3.1, we download a new version of cmake and use it when configuring projects that use lcm.

**Comment by patmarion** [comment]
i made the required fix in `master` so this PR would pass on travis-ci ubuntu14.  thanks!
